team-telnyx/webrtc
# refactor: convert js to ts

Fixes the following build error on import in Angular projects:
```shell
ERROR in node_modules/@telnyx/webrtc/lib/BaseClient.d.ts:1:26 - error TS2307: Cannot find module './TypedEmitter'.

1 import TypedEmitter from './TypedEmitter';
                           ~~~~~~~~~~~~~~~~
node_modules/@telnyx/webrtc/lib/Verto/Verto.d.ts:1:28 - error TS2307: Cannot find module './LiveArray'.

1 import VertoLiveArray from './LiveArray';
                             ~~~~~~~~~~~~~
node_modules/@telnyx/webrtc/lib/Verto/Verto.d.ts:2:23 - error TS2307: Cannot find module './ConfMan'.

2 import VertoConf from './ConfMan';
                        ~~~~~~~~~~~
node_modules/@telnyx/webrtc/lib/Verto/Verto.d.ts:3:25 - error TS2307: Cannot find module './Dialog'.

3 import VertoDialog from './Dialog';
```

See https://github.com/microsoft/TypeScript/issues/7546#issuecomment-535699638

## Manual testing

1. Run `npm link && npm run build`
2. Outside of this repo, generate new project with [Angular CLI](https://github.com/angular/angular-cli) and install `@telnyx/webrtc`
3. From the Angular project, run `npm link @telnyx/webrtc`
4. Import `TelnyxRTC` and start app with `ng serve`. App should compile without errors

**Warning** I haven't tested actual calls as of 4d59d0f28a03f30e8257ff55c1a51fff914134d0
